The test failure seems relevant:

```
[INFO] Running io.quarkus.maven.it.DevMojoIT
[ERROR] Tests run: 14, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 296.08 s <<< FAILURE! - in io.quarkus.maven.it.DevMojoIT
[ERROR] testThatTheApplicationIsReloadedMultiModule  Time elapsed: 79.409 s  <<< ERROR!
org.awaitility.core.ConditionTimeoutException: Condition with io.quarkus.maven.it.DevMojoIT was not fulfilled within 1 minutes.
	at io.quarkus.maven.it.DevMojoIT.testThatTheApplicationIsReloadedMultiModule(DevMojoIT.java:119)
Caused by: java.util.concurrent.TimeoutException
	at io.quarkus.maven.it.DevMojoIT.testThatTheApplicationIsReloadedMultiModule(DevMojoIT.java:119)

[ERROR] testThatNewResourcesAreServed  Time elapsed: 73.058 s  <<< ERROR!
org.awaitility.core.ConditionTimeoutException: Condition with io.quarkus.maven.it.DevMojoIT was not fulfilled within 1 minutes.
	at io.quarkus.maven.it.DevMojoIT.testThatNewResourcesAreServed(DevMojoIT.java:339)
Caused by: java.util.concurrent.TimeoutException
	at io.quarkus.maven.it.DevMojoIT.testThatNewResourcesAreServed(DevMojoIT.java:339)
```

--- a/core/devmode/src/main/java/io/quarkus/dev/RuntimeUpdatesProcessor.java
+++ b/core/devmode/src/main/java/io/quarkus/dev/RuntimeUpdatesProcessor.java
@@ -39,6 +39,12 @@ public class RuntimeUpdatesProcessor implements HotReplacementContext {
     private volatile Map<String, Boolean> watchedFilePaths = Collections.emptyMap();
     private final Map<Path, Long> watchedFileTimestamps = new ConcurrentHashMap<>();
     private final Map<Path, Path> classFilePathToSourceFilePath = new ConcurrentHashMap<>();
+    /**
+     * resources that appear in both src and target, these will be removed if the src resource subsequently disappears
+     *
+     * This set contains the paths in the target dir
+     */
+    private final Set<Path> correspondingResources = Collections.newSetFromMap(new ConcurrentHashMap<>());
 
     private static final Logger log = Logger.getLogger(RuntimeUpdatesProcessor.class.getPackage().getName());
     private final List<Runnable> preScanSteps = new CopyOnWriteArrayList<>();
@@ -272,6 +278,47 @@ public class RuntimeUpdatesProcessor implements HotReplacementContext {
             }
             Path root = Paths.get(rootPath);
             Path classesDir = Paths.get(module.getClassesPath());
+            //copy all modified non hot deployment files over
+            if (doCopy) {
+                try {
+                    Set<Path> seen = new HashSet<>(correspondingResources);
+                    Files.walk(root).forEach(new Consumer<Path>() {
+                        @Override
+                        public void accept(Path path) {
+                            try {
+                                Path relative = root.relativize(path);
+                                Path target = classesDir.resolve(relative);
+                                seen.remove(target);
+                                if (!watchedFileTimestamps.containsKey(path)) {
+                                    correspondingResources.add(target);
+                                    if (!Files.exists(target) || Files.getLastModifiedTime(target).toMillis() < Files
+                                            .getLastModifiedTime(path).toMillis()) {
+                                        if (Files.isDirectory(path)) {
+                                            Files.createDirectories(target);
+                                        } else {
+                                            Files.createDirectories(target.getParent());
+                                            byte[] data = Files.readAllBytes(path);
+                                            try (FileOutputStream out = new FileOutputStream(target.toFile())) {
+                                                out.write(data);
+                                            }
+                                        }
+                                    }
+                                }
+                            } catch (Exception e) {
+                                log.error("Failed to copy resources", e);
+                            }
+                        }
+                    });
+                    for (Path i : seen) {
+                        if (!Files.isDirectory(i)) {
+                            Files.delete(i);
+                        }
+                    }
+                } catch (IOException e) {
+                    log.error("Failed to copy resources", e);
+                }
+
+            }
 
             for (String path : watchedFilePaths.keySet()) {
                 Path file = root.resolve(path);

--- a/extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/sql_load_script/AddNewSqlLoadScriptTestCase.java
+++ b/extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/sql_load_script/AddNewSqlLoadScriptTestCase.java
@@ -1,0 +1,31 @@
+package io.quarkus.hibernate.orm.sql_load_script;
+
+import org.hamcrest.Matchers;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.hibernate.orm.MyEntity;
+import io.quarkus.test.QuarkusDevModeTest;
+import io.restassured.RestAssured;
+
+public class AddNewSqlLoadScriptTestCase {
+    @RegisterExtension
+    static QuarkusDevModeTest runner = new QuarkusDevModeTest()
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
+                    .addAsResource("application.properties")
+                    .addClasses(SqlLoadScriptTestResource.class, MyEntity.class));
+
+    @Test
+    public void testAddNewImportSql() {
+        String name = "default sql load script entity";
+        RestAssured.when().get("/orm-sql-load-script/1").then().body(Matchers.is(name));
+
+        runner.modifyResourceFile("application.properties",
+                (s) -> s += "\nquarkus.hibernate-orm.sql-load-script=new-load-script-test.sql");
+        runner.addResourceFile("new-load-script-test.sql", "INSERT INTO MyEntity(id, name) VALUES(1, 'NEW SCRIPT');");
+        RestAssured.when().get("/orm-sql-load-script/1").then().body(Matchers.is("NEW SCRIPT"));
+
+    }
+}

--- a/integration-tests/maven/src/test/java/io/quarkus/maven/it/DevMojoIT.java
+++ b/integration-tests/maven/src/test/java/io/quarkus/maven/it/DevMojoIT.java
@@ -22,6 +22,10 @@ import io.quarkus.maven.it.verifier.RunningInvoker;
 
 /**
  * @author <a href="http://escoffier.me">Clement Escoffier</a>
+ *
+ *         NOTE to anyone diagnosing failures in this test, to run a single method use:
+ *
+ *         mvn install -Dit.test=DevMojoIT#methodName
  */
 public class DevMojoIT extends RunAndCheckMojoTestBase {
 

--- a/integration-tests/maven/src/test/java/io/quarkus/maven/it/GenerateConfigIT.java
+++ b/integration-tests/maven/src/test/java/io/quarkus/maven/it/GenerateConfigIT.java
@@ -63,7 +63,6 @@ class GenerateConfigIT extends MojoTestBase {
             throws MavenInvocationException, FileNotFoundException, UnsupportedEncodingException {
         InvocationRequest request = new DefaultInvocationRequest();
         request.setBatchMode(true);
-        request.setDebug(true);
         request.setGoals(Collections
                 .singletonList(CreateProjectMojo.PLUGIN_KEY + ":" + MojoUtils.getPluginVersion() + ":generate-config"));
         Properties properties = new Properties();

--- a/test-framework/junit5-internal/src/main/java/io/quarkus/test/QuarkusDevModeTest.java
+++ b/test-framework/junit5-internal/src/main/java/io/quarkus/test/QuarkusDevModeTest.java
@@ -314,7 +314,7 @@ public class QuarkusDevModeTest
         sleepForFileChanges(path);
     }
 
-    public void modifyFile(String name, Function<String, String> mutator, Path path) {
+    void modifyFile(String name, Function<String, String> mutator, Path path) {
         try (Stream<Path> sources = Files.walk(path)) {
             sources.forEach(s -> {
                 if (s.endsWith(name)) {

